hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/api/DateMatcher.kt

```diff
@@ -47,18 +48,21 @@ data class DateMatcher(
     }
   }
 
-  override fun buildQuery(): AbstractQueryBuilder<*> = when {
-    minValue != null && maxValue != null -> {
-      QueryBuilders.rangeQuery(attribute).from(minValue).includeLower(minInclusive).to(maxValue).includeUpper(maxInclusive)
-    }
-    minValue != null -> {
-      QueryBuilders.rangeQuery(attribute).from(minValue).includeLower(minInclusive)
-    }
-    maxValue != null -> {
-      QueryBuilders.rangeQuery(attribute).to(maxValue).includeUpper(maxInclusive)
-    }
-    else -> throw AttributeSearchException("Attribute $attribute must have a min or max value")
-  }
+  override fun buildQuery(attributes: Attributes): AbstractQueryBuilder<*> =
+    attributes[attribute]?.let {
+      when {
+        minValue != null && maxValue != null -> {
+          QueryBuilders.rangeQuery(it.openSearchName).from(minValue).includeLower(minInclusive).to(maxValue).includeUpper(maxInclusive)
```

Comment:
In all our queries we now reference the `openSearchName` which has the keyword suffix if required.